### PR TITLE
[WIP] Tutorial PR

### LINKED ISSUE #174: SciPy 2019 tutorial plan
This issue collects to-do items and planning for the SciPy 2019 tutorials, subsuming previous such issues #42, #90, and #97. Bigger tasks we expect to do before the tutorial:

- [x] Introduction 01 should be replaced with a short demo (10-15min) (e.g. datashader 1billion dashboard, gapminders, nb with other types of plots) [#188]
- [x] Move Introduction 01 out of the tutorial itself, to be next to background nb as an overview (jsignell) [#188]
- [x] Reduce total amount of material expected to be covered

NEW WRITING:
- [x] Boost focus on Panel
- [ ] Boost coverage of libraries other than Bokeh
- [x] More/better/more-hands-on exercises
- [x] Reduce overlap between tutorial sections
- [x] Try to make the difference between holoviews and hvplot more explicit in the text.
- [ ] Review feedback from prior tutorials (incorporated below)
    * [ ] Add list of references and weblinks for further exploration
    * [ ] Have a list of conference presentations coming up that use these tools (@jbednar)
    * [ ] Explain what these packages will _not_ do (@jbednar)
    * [ ] How do other tools fit in? Seaborn, ggpy, etc.  (@jbednar)
    * [x] Early examples were basic, then later ones quickly became impenetrable, so that we could only see what CAN be done, not HOW to do it.  Try to balance better.

ENV MANAGEMENT:
- [x] Move topics from datashader and holoviews sites into pyviz, as most combine multiple tools and make good launching-off points. (jsignell)
- [x] Streamline conda environment instructions/setup: (jsignell)
    * [x] Only install holoviews and not geoviews
    * [x] Remove conda-forge from list of channels
    * [x] Update pinning, if needed
    * [x] Measure the time to build the conda environment, and see if it can be improved if not reasonable
    * [ ] Test conda install on Windows
    * [x] Make sure everything is pip-installable as well as conda-installable (#81) (but warn pip users that they are on their own)

Other to-do items we hope to be able to get to as well:

- [x] Remove numbers from the website versions of notebooks, leaving them in the .ipynb files to show ordering (jsignell)
- [x] Remove ! and > characters in the prompts in the installation instructions; Windows users (in particular) were copy-pasting this into their command prompt (jsignell)
- [ ] Provide a binder or similar environment that users can use if their environment doesn't work (https://github.com/pyviz/pyviz/issues/98)
- [ ] Separate suggested diversions/more detail more clearly from what to do next in the tutorial.
- [x] Add multiple plotting libraries to the first tutorial sections
- [ ] How to handle matplotlib inline and discuss it
- [x] Make sure data isn't redownloaded when updating, when following our instructions
- [ ] Merge in content from the PyData Cookbook (Jim; https://github.com/pydata/pydata-cookbook/pull/26)
- [ ] Change 08_Custom_Interactivity.ipynb to use new RangeSelect tool
- [x] Mention new Links feature that will allow custom interactivity without needing a live server
- [x] Introduce DynamicMap and HoloMap properly
- [x] 09_Operations_and_Pipelines.ipynb starts using datashader without introducing it/why to use it/that it will be described in the next section?
- [x] Tighten up descriptions -- Q: is the whole thing a tutorial, or is each notebook a tutorial? A: Each nb is a section
- [x] Reduce overlap with holoviews.org, primarily just pointing people from pyviz.org to holoviews.org
- [x] Should fold in tutorial A1's material somewhere; it's odd to have it in an appendix. It made sense when the tutorial was smaller, but now that there is always more material than can be covered, it's confusing to have it separate, so that the other material can't tell whether these concepts have been introduced.

Seemingly unrelated tasks mentioned in previous issues, collected here to avoid forgetting them:

- [ ] Add a section to pyviz.org with links to maintainer-focused projects (nbsite, nbval, nbsmoke, autover, etc.)
- [ ] Make pyviz tests cover installing as a package and other new functionality (#81)
- [ ] Add twitter feed listening to multiple projects
- [ ] Need automated scripts for making at least the first pass at the site logo and favicon bitmaps from a transparent SVG source; currently edited those by hand for pyviz.org and other project sites have non-transparent logos. (See pyviz/nbsite#40)

### PR DESCRIPTION
**Pull request by jsignell**
This is the branch on which we can write the new tutorial. I have written a very preliminary outline. in index.ipynb

Closes #174

### CODE REVIEW

**Comment by jsignell** [comment]
**NOTE:** If you change the name of a notebook: you need to change the links in:
- the notebook's first cell (header)
- examples/tutorial/index.ipynb
- doc/tutorial/index.rst
- in any notebooks that link to it

File: examples/tutorial/06_Custom_Interactivity.ipynb [hunk 1]
```diff
@@ -550,7 +550,7 @@
    "source": [
     "def histogram_callback(index):\n",
     "    if len(index) == 0:\n",
-    "        return  hv.Histogram((None, None), 'longitude', 'latitude')\n",
+    "        return  hv.Histogram(([], []), kdims=['longitude'], vdims=['latitude'])\n",
```
Comment (philippjfr) [inline examples/tutorial/06_Custom_Interactivity.ipynb]: A Histogram with longitude and latitude as the dimensions?

Comment (philippjfr) [reply examples/tutorial/06_Custom_Interactivity.ipynb]: Also an empty list is the usual empty constructor, i.e.

```
hv.Histogram([])
```

**Comment by jlstevens** [comment]
@jbednar @philippjfr Managed to get the [dev website built](https://pyviz-dev.github.io/holoviz/tutorial/index.html) at this (incomplete) stage of the PR. Should we now merge to update the main public website with this work in progress?

**Comment by jbednar** [comment]
I think we should update the main website so that people can start following the new instructions, but merging would be confusing as the PR isn't actually complete yet. Can't you just push a tag 'website' to this branch to make the main website update? I guess if we have to merge, then do, and open a new PR, but it would be nicer to stay in this PR and just keep tagging it whenever we are ready for a website update.

**Pull request merged**